packages/atomic/src/components/commerce/product-template-components/atomic-product-text/atomic-product-text.tsx

```diff
@@ -33,11 +37,39 @@ export class AtomicProductText
    * This will look in the Product object first, and then in the product.additionalFields object for the fields.
    */
   @Prop({reflect: true}) public field!: string;
+  /**
+   * When this is set to `true`, the component attempts to highlight text based on the highlighting properties provided by the search API response.
+   * This property only works for the document excerpt and the ec_name field.
+   */
+  @Prop({reflect: true}) public shouldHighlight = true;
   /**
    * The locale key for the text to display when the configured field has no value.
    */
   @Prop({reflect: true}) public default?: string;
 
+  private renderWithHighlights(
```

Comment:
This could be extracted in a utility function in `/common` (or create functional component(s) for `atomic-result-text` and `atomic-product-text` since there is more overlap for the warning on invalid field). 

 To make sure there is no `@coveo/headless/commerce` import in the `/common` folder, `HighlightUtils.highlightString` can be abstracted into an interface in that utility function.

Reply:
I think the second option makes more sense here